File: python/advent/common/file_utils.py
from pathlib import Path


_DATA_DIRECTORY = Path(__file__).parent.parent.parent.parent / "_data"


def read_challenge_input_lines(file_name: str):
    file_path = _DATA_DIRECTORY / file_name

    with file_path.open('r') as f:
        while True:
            line = f.readline()
            yield line.strip()

            if not line.endswith("\n"):
                break

File: fsharp/advent/common/file_utils.fs
namespace BeardedPlatypus.advent.common

open PathLib

module internal file_utils =
    let internal read_input_file (file_name: string) : string seq =
        let file_path = Paths.Create("resources") / file_name
        System.IO.File.ReadLines(file_path.ToString())
        
